federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -132,14 +158,45 @@ func NewIngressController(client federation_release_1_4.Interface) *IngressContr
 
 		&util.ClusterLifecycleHandlerFuncs{
 			ClusterAvailable: func(cluster *federation_api.Cluster) {
-				// When new cluster becomes available process all the ingresses again.
+				// When new cluster becomes available process all the ingresses again, and configure it's ingress controller's configmap with the correct UID
 				ic.clusterDeliverer.DeliverAt(allClustersKey, nil, time.Now().Add(ic.clusterAvailableDelay))
 			},
 		},
 	)
 
-	// Federated updater along with Create/Update/Delete operations.
-	ic.federatedUpdater = util.NewFederatedUpdater(ic.ingressFederatedInformer,
+	// Federated informer on configmaps for ingress controllers in members of the federation.
+	ic.configMapFederatedInformer = util.NewFederatedInformer(
+		client,
+		func(cluster *federation_api.Cluster, targetClient kube_release_1_4.Interface) (cache.Store, framework.ControllerInterface) {
+			return framework.NewInformer(
+				&cache.ListWatch{
+					ListFunc: func(options api.ListOptions) (pkg_runtime.Object, error) {
+						return targetClient.Core().ConfigMaps(uidConfigMapNamespace).List(options) // Todo - we only want to list one by name - need options to reflect that.
+					},
+					WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+						return targetClient.Core().ConfigMaps(api.NamespaceSystem).Watch(options) // TODO: As above
+					},
+				},
+				&api.ConfigMap{},
+				controller.NoResyncPeriodFunc(),
+				// Trigger reconcilation whenever the ingress controller's configmap in a federated cluster is changed. In most cases it
+				// would be just confirmation that the configmap for the ingress controller is correct.
+				util.NewTriggerOnAllChanges(
+					func(obj pkg_runtime.Object) {
+						ic.reconcileConfigMapForCluster(cluster.Name)
```

Comment:
Reconcilation should happen on a separate thread. Reconciliating here you will block the whole informer.